variantlib/plugins/_subprocess.py

```diff
@@ -71,41 +93,137 @@ def process_configs(
     return [{"name": vfeat.name, "values": vfeat.values} for vfeat in configs]
 
 
+def process_dynamic_configs(
+    configs: dict[VariantFeatureName, list[VariantFeatureValue]],
+    plugin_instance: PluginDynamicType,
+    method: str,
+) -> list[dict[str, str | list[str]]]:
+    try:
+        validate_type(configs, dict[VariantFeatureName, list[VariantFeatureValue]])
+
+    except ValidationError as err:
+        raise TypeError(
+            f"Provider {plugin_instance.namespace}, {method}() "
+            f"method returned incorrect type. {err}"
+        ) from None
+
+    return [
+        {"name": name, "values": values} for name, values in configs.items() if values
+    ]
+
+
+@dataclass(frozen=True)
+class VariantProperty:
+    namespace: str
+    feature: str
+    value: str
+
+    @classmethod
+    def from_str(cls, input_str: str) -> Self:
+        # Try matching the input string with the regex pattern
+        match = VALIDATION_PROPERTY_REGEX.fullmatch(input_str.strip())
```

Comment:
The whole point of subprocess is to keep validation in the parent, and pass only validated data.